BLSQ/iaso
# IA-2937 update filters and tables

Explain what problem this PR is resolving

Related JIRA tickets : IA-2937

## Self proofreading checklist


## Doc
Added comments in the code

## Changes

- Updated redirections in all tables and filters:
    - removed redux (`redirectTo`/`redirectToReplace`)
    - replaced with `useRedirectTo`/`useRedirectToReplace`
- Updated `SingleTable` (same change)
- Updated `useFilterState`(idem)
- Updated shared `Filter` component
- Improved `useParamsObject` to better handle ambiguous urls
- Fixed Completeness Stats page crash by removing reference to `router`
- Updated `TableWithDeepLink`'s redirection
- Removed useless `onTableParamsChange` in `TableWithDeepLink` instances
- Improved some API calls by adding `staleTime`, `cacheTime` and `keepPreviousData` to the react-query hook


## How to test

Update package.json to use branch `IA-2941_refactor_IconButton` from bluesquare-components

Try out all filters and tables in iaso:
- urls should be updated and the API called when changing sort, page or pageSize in a Table
- Same when changing filter value

Note:
- Don't forget to run the worker to generate potential payments


## Notes

- Depends on https://github.com/BLSQ/bluesquare-components/pull/149
- Total removal of redux's `redirectTo` and `redirectToReplace` will be done in next PR
